runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -913,11 +901,24 @@ int32_t TR_UnsafeFastPath::perform()
                TR::Node *addrCalc = NULL;
 
                // Calculate element address
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+               if (isArrayOperation && TR::Compiler->om.isOffHeapAllocationEnabled())
+                  {
+                  TR::Node *baseNodeForAdd = TR::TransformUtil::generateDataAddrLoadTrees(comp(), base);
+                  TR::Node *newOffset = TR::Node::create(TR::ladd, 2, offset, TR::Node::lconst(-TR::Compiler->om.contiguousArrayHeaderSizeInBytes()));
```

Comment:
two concerns here:
1) this is dealing with unsafe and we expect the in-coming offset doesn't contain off-heap array header already. doing a minus here might be a problem;
2) this could be better if the offset is a compile-time constant. you don't need the ladd op;

Reply:
1. Correct, fixed
2. True, which will work for offheap and non-offheap, I'll open an issue for that 👍